tests/test_views.py

```diff
@@ -0,0 +1,345 @@
+        self.assertTrue(self.client.login(username="testuser1", password="123"))
+        response = self.client.post(self.url)
+        self.assertEqual(200, response.status_code)
+        self.assertIn("plan", response.context["form"].errors)
+        self.assertIn("This field is required.", response.context["form"].errors["plan"])
+
+    @patch("stripe.Customer.create", return_value=PropertyMock(id="cus_xxx1234567890_02"))
+    def test_post_new_sub_no_proration(self, stripe_customer_mock):
+        self.assertTrue(self.client.login(username="testuser2", password="123"))
+        response = self.client.post(self.url, {"plan": "test0"})
+        self.assertEqual(200, response.status_code)
+        self.assertIn("form", response.context)
+        self.assertIn("You must already be subscribed to a plan before you can change it.", response.context["form"].errors["__all__"])
+
+    @patch("djstripe.models.Customer.current_subscription", new_callable=PropertyMock, return_value=CurrentSubscription(plan=test_plan, amount=Decimal(25.00)))
+    @patch("djstripe.models.Customer.subscribe", autospec=True)
+    def test_change_sub_no_proration(self, subscribe_mock, current_subscription_mock):
+        self.assertTrue(self.client.login(username="testuser1", password="123"))
+        response = self.client.post(self.url, {"plan": "test0"})
+        self.assertRedirects(response, reverse("djstripe:history"))
+
+        subscribe_mock.assert_called_once_with(self.user1.customer, "test0")
+
+    @patch("djstripe.views.PRORATION_POLICY_FOR_UPGRADES", return_value=True)
+    @patch("djstripe.models.Customer.current_subscription", new_callable=PropertyMock, return_value=CurrentSubscription(plan=test_plan, amount=Decimal(25.00)))
+    @patch("djstripe.models.Customer.subscribe", autospec=True)
+    def test_change_sub_with_proration_downgrade(self, subscribe_mock, current_subscription_mock, proration_policy_mock):
+        self.assertTrue(self.client.login(username="testuser1", password="123"))
+        response = self.client.post(self.url, {"plan": "test0"})
+        self.assertRedirects(response, reverse("djstripe:history"))
+
+        subscribe_mock.assert_called_once_with(self.user1.customer, "test0")
+
+    @patch("djstripe.views.PRORATION_POLICY_FOR_UPGRADES", return_value=True)
+    @patch("djstripe.models.Customer.current_subscription", new_callable=PropertyMock, return_value=CurrentSubscription(plan=test_plan, amount=Decimal(25.00)))
+    @patch("djstripe.models.Customer.subscribe", autospec=True)
+    def test_change_sub_with_proration_upgrade(self, subscribe_mock, current_subscription_mock, proration_policy_mock):
+        self.assertTrue(self.client.login(username="testuser1", password="123"))
+
+        response = self.client.post(self.url, {"plan": "test2"})
+        self.assertRedirects(response, reverse("djstripe:history"))
+
+        subscribe_mock.assert_called_once_with(self.user1.customer, "test2", prorate=True)
```

Comment:
In this case, prorate=True is not explicitly called and the default argument gets used. Just take that out and this test will pass.